src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -201,48 +206,33 @@ public static JpegFileMarker FindNextFileMarker(BufferedReadStream stream)
         /// <inheritdoc/>
         public Image<TPixel> Decode<TPixel>(BufferedReadStream stream, CancellationToken cancellationToken)
             where TPixel : unmanaged, IPixel<TPixel>
-            => this.Decode<TPixel>(stream, targetSize: null, cancellationToken);
-
-        /// <inheritdoc/>
-        public IImageInfo Identify(BufferedReadStream stream, CancellationToken cancellationToken)
         {
-            this.ParseStream(stream, spectralConverter: null, cancellationToken);
+            using var spectralConverter = new SpectralConverter<TPixel>(this.configuration, this.Options.GeneralOptions.TargetSize);
```

Comment:
It would be very important to have some tests validating how resize works together with #2076, stressing different dimensions including ones that are not divisible by 8. /cc @br3aker

Reply:
Shouldn't be too hard as reference results are just scaled input images with box filter applied. I can create a PR when actual API is set in stone - you gave a lot to think of (in a positive way)!

Reply:
Thanks @br3aker 👍

I didn't add any specific ones because we had discussed the output quality vs performance tradeoffs in #2076 . Output seemed adequate and perf was far too negatively affected when using other resamplers.